0.27

balloob opened the pull request:
#0.27 is here to break eggs and take names: notifications, Hue fakery, safety and unification come to Home Assistant

This week's blog post could only be possibly described by exactly one word:
# #Amazing

<sup>or <sup>maybe<sup>#supersized</sup></sup></sup>

Keep reading to see what **#Amazing** things we have in store for you this week :smile:! And make sure you read all the way to the end, because I left a present down there for those committed few among you :)

But first...
## Some general housekeeping

Paulus (@balloob) is on vacation in Europe this week, so you will all have to deal with me, Robbie (@robbiet480) for this release blog post. Don't worry, Paulus will be back to tearing apart your pull requests in no time :-). 

Special thanks to my awesome helpers for this week's release who are looking over my shoulder to make sure I'm crossing my t's and dotting my i's: @Teagan42, @infamy and @fabaff.

For my next trick, let's hand out some...
## Trophies

I felt that I had to 1-up Paulus (@balloob) somehow with his 500,000 pageviews stat he shared in the [0.26 blog post](https://home-assistant.io/blog/2016/08/13/foursquare-fast-com-ffmpeg-gpsd/), so I pushed myself and our development community as a whole super hard the last two weeks to put a lot of love into Home Assistant to bring you not just 1, but 6 **#Amazing** stats for this release. As of 0.27, we have now surpassed the following milestones:
- [10,000](https://travis-ci.org/home-assistant/home-assistant/builds/154660811) builds on Travis (congrats to @BluGeni)
- [3,000](https://github.com/home-assistant/home-assistant/pull/3000) issues and pull requests (thanks @kellerza!)
- [2,000](https://github.com/home-assistant/home-assistant/pull/2991) pull requests alone (awesome @fabaff!!)
- 900 forks on Github!

In addition,
- We are very close to 4,000 stars on Github!
- [I'm now #5 contributor by most commits!!!](https://github.com/home-assistant/home-assistant/graphs/contributors) Hey wait, how'd this get in here 😳 

Now that we have that great news out of the way, onto this week's release which is going to keep the **#Amazing** gravy train rolling right along and get to the stuff you all _really_ are here for.
## 0.27

While this release is **#Amazing**, we had to break a few eggs (now you understand the title reference!) to make a beautiful omelette (using home automation obviously) so some platforms and components have needed to introduce breaking changes. Please make sure to read the Breaking Changes section below.
#### [Hue Bridge Emulation](https://home-assistant.io/components/emulated_hue/)

Thanks to @mgbowen we now have the functionality previously provided by @blocke's [ha-local-echo](https://github.com/blocke/ha-local-echo) built right into Home Assistant! This means that for those of you with devices that either lack or have a subpar integration with Home Assistant (looking at you Amazon Echo) you can now have a better experience by having your Home Assistant pretend to be a Hue Bridge. Personally, I have used @auchter's [Haaska](https://github.com/auchter/haaska) previously but found that it was slow to respond and sometimes failed entirely. With the new [`emulated_hue`](https://home-assistant.io/components/emulated_hue/) component, you can have local control of entities through Amazon Echo.
#### Notification improvements

We have some excellent upgrades to the notification system coming to you in 0.27, courtesy of me, @robbiet480.
##### [HTML5 Push Notifications](https://home-assistant.io/components/notify.html5/)

This release adds support for HTML5 push notifications on Chrome/Firefox/Opera on both desktop and Android devices. This means that you can send a notification to your phone even your Home Assistant is not in your mobile browser. When using Chrome you can even include 2 action buttons so that you can control your Home Assistant from your phone's lock screen, allowing you to do things like sound alarms or unlock your front door, all without leaving the notification. Thanks again to me (@robbiet480) and Paulus (@balloob) for all the hard work on this!

![screenshot_20160816-225335](https://cloud.githubusercontent.com/assets/1444314/17917008/3a58dab0-696d-11e6-9c40-4f47c727e142.png)
##### [Notification Groups](https://home-assistant.io/components/notify.group/)

Using the new notify `group` platform allows you to cut down a lot of duplicate automation logic by combining multiple notification platforms and `target`s into a single notify service. Check out the [docs](https://home-assistant.io/components/notify.group/) for more info.
##### `target` is no longer needed

For platforms that support it, starting with the new HTML5 platform, any `target`s that are available will be exposed as individual services, so no more having to remember which `target`s to use. Please note that the existing services also still exist so you can keep using `target` if you wish.
#### Validate config before a restart

Ever restarted Home Assistant to test a config change just to find out there is a validation error? Well, not anymore! @kellerza has added a command line script that will validate your configuration as if you started Home Assistant.

``` bash
$ hass --script check_config
```
#### Config validation

This release includes a big push on making sure all platforms contain proper configuration validation. This should help in getting your configuration right. Thanks to @fabaff, @pavoni, @pvizeli, @nkgilley for all the hard work on this, you all rock!

![screen shot 2016-08-23 at 8 20 41 pm](https://cloud.githubusercontent.com/assets/1444314/17917210/22ede3aa-696f-11e6-84f0-f7a7653cd695.png)
#### [FFMPEG motion / noise sensing](https://home-assistant.io/components/binary_sensor.ffmpeg/)

It's now possible to use FFMPEG to monitor a video stream and detect motion thanks to a new binary sensor platform by @pvizeli.
#### Component clean up - `thermostat` & `hvac` -> `climate`. `rollershutter` & `garage_door` -> `cover`

Due to our wild growth we ended up with a few components that had a lot of overlapping functionality. @turbokongen took on the hard job on merging them. Thermostat and HVAC platforms are now combined under the new Climate component. Rollershutter and Garage Door platforms are now combined under the new Cover component. You can easily upgrade by just swapping out the name. For example replace `thermostat` with `climate`. The old components have been deprecated and will be removed in the near future.
#### A new `fan` component

Along with the new `climate` component, @Teagan42 and I (@robbiet480) decided we needed something simpler to just control a fan. Currently it has support for controlling Insteon fans. MQTT support will appear in 0.28.0. I tried to get it implemented before 0.27.0 but spent too long writing this blog post 😢.
#### All changes
- Merge thermostat and HVAC components into new climate component (@turbokongen)
- Merge rollershutter and garage door components into new cover component (@turbokongen)
- Alarm Control Panel - Manual: Allow returning to previous state after trigger (@tobiebooth)
- Sensor - DHT: Allow range checking (@open-homeautomation)
- Light entities will now expose their features (@shmuelzon)
- Sensor: Monitor HP ILO sensors on HP servers (@Juggels)
- Sensor: Monitor Fritzbox Calls (@DavidMStraub)
- Notify: LlamaLab Automate is now supported (@danielperna84)
- Sensor: Serial CO2 sensors now supported (@open-homeautomation)
- Sensor: MQTT room presence detection (@mKerix)
- Notify: New group platform allows grouping notify targets across platforms (@robbiet480)
- HTML5 push notifications (@robbiet480, @balloob)
- Weather: Wunderground now supported (@arsaboo, @Teagan42)
- New check config script to test validity before restarting HA (@kellerza)
- Binary Sensor: Allow monitoring a camera feed using FFMpeg (@pvizeli)
- Cover: Z-Wave platform now supports positions (@nunofgs)
- Device tracker: allow using Gravatar for entity picture (@robbiet480)
- Notify: platforms with known targets will expose them as standalone services (@robbiet480)
- Camera: MJPEG and Generic camera's can now authenticate using digest auth (@meatz)
- Weather: Forecast.io now can show daily temp/precip forecast values (@DavidMStraub)
- Media Player: WebOS TV now allows customizing the sources (@roidayan)
- Device tracker: Allow tracking devices using Bluetooth Low-Energy (@open-homeautomation)
- Notify: Ensure Slack messages appear as correct user (@technicalpickles)
- YAML: Secrets will look for values in all parents folders up to the config root folder (@Teagan42)
- Camera: Generic camera now supports template support (@balloob)
- Slack: Allow sending attachments (@technicalpickles)
- Device Tracking: TP-Link Archer C7 5ghz devices will now also be found (@dpford)
- Z-Wave: New rename node service added (@jnewland)
- Wink: bug fixes (@w1ll1am23)
- Homematic: new device support, add controller state and variable support (@danielperna84 @pvizeli, @mcdeck)
- Allow emulating a Hue bridge to control entities (@mgbowen)
#### Breaking changes
- Ecobee3 occupancy sensors have moved from sensor platform to binary sensor platform.
- Forecast.io entity IDs are now like `sensor.forecastio_temperature`. Previously they were like `sensor.weather_temperature`. Apologies for this change, but we needed to make Forecast.io more generic now that we have many weather platforms.
- The loopenergy sensor configuration format changed slightly, please reformat based on the revised documentation.
- The configuration for the Sabnzbd sensor has slightly changed. The prefix `type:` is no longer required for monitored variables.
#### Deprecations
- Using the `thermostat` and `hvac` components has been deprecated. Please migrate to the new `climate` component. (just change the component name, the configurations are compatible)
- Using the `rollershutter` and `garage_door` components have also been deprecated. Please migrate to the new `cover` component. (just change the component name, the configurations are compatible)
## Finishing up

Thanks all for sticking with me to the end. I'll be taking over a lot of Paulus's (@balloob) work while he is gone, but as I said, don't worry because he'll be back well before 0.28 comes out. Hopefully you didn't find this jovial blog post too jarring from our standard style, I just wrote a lot of this at 2am after being awake for almost 20 hours, so I'm a little loopy hahaha 😴.

Also, thanks as always to our developer contributors, documentation contributors, but most of all our users! This would've just been a script that Paulus (@balloob) used to control his lights at home if we didn't have your enthusiasm.

Feel free to let me know what you thought of this blog post and release on Gitter or my [Twitter](https://twitter.com/robbie), or even the [Home Assistant Twitter](https://twitter.com/home_assistant). Did I mention we have a brand new [Facebook page](https://www.facebook.com/homeassistantio) that you should absolutely Like? There's a convenient Facebook Like and Twitter follow button right over there -->.

I almost forgot about your 🎁 for reading all the way to here: a :cookie:! Hope you enjoy it in good health :smile:.

Talk to you soon on Gitter and in your pull request comments!
- Robbie

(p.s. To those of you that scrolled directly to the bottom to get your present, just know that you didn't earn it like the others did. :smile:)

MartinHjelmare committed:
Add MQTT gateway for MySensors

* Use mqtt component to enable a MySensors MQTT gateway.
* Setup the MQTT gateway if mysensors config has mqtt as a value for
	the key	device in the list of gateways.
* Simplify two lines in the mqtt component.

Commit:
Return to previous alarm state after trigger (#2580)

tobiebooth committed:
add disarm_after_trigger to manual alarm panel

balloob committed:
Version bump to 0.27.0.dev0

Commit:
Bugfix: removed conf_platform (#2811)

* Bugfix: removed conf_platform

* Remove unused import

* Fix for wrong update

Commit:
Implemented range checking for temperature and humidity. Out-of-range… (#2805)

* Implemented range checking for temperature and humidity. Out-of-range values will be ignored

* Removed unused import

* Use celsius_to_fahrenheit conversion method

balloob committed:
Html5 push notifications notify platform (#2807)

* Initial work to add Chrome Push Notification support

* Remove push.js from home-assistant since it is now in Polymer

* Chrome->HTML5, general cleanup/fixes

* Make html5 generic, move manifest.json into frontend so that we can dynamically add the gcm_sender_id

* Pylint, flake8, pydocstyle frontend init

* HTML5 push fixes

* Update polymer

* Remove crypto req

* Add notify default platform.

* Fix HTML5 push

* Registration fixes

* Linting fix

* pep257 fix

* Add tests

* pep257 fix

* Update frontend

turbokongen committed:
Fix unknown unit of measurement for hvac and thermostat component (#2816)

* Fix unknown unit of measurement for hvac and thermostat component

* Simplify

mKeRix committed:
Check for existence of system mode on Honeywell thermostats (#2815)

* Check for existence of system mode on Honeywell thermostats

* Return None instead of undefined

* Use getattr instead of if/else

pavoni committed:
Bump to pywemo 0.4.5 - fixes bug with requests 2.11.0 (#2818)

pavoni committed:
Handle accuracy zero correctly in enter/leave events.

pavoni committed:
Add missed docstring.

balloob committed:
Merge pull request #2734 from tobiebooth/manual-alarm-improvements

Return to previous alarm state after trigger (#2580)

MartinHjelmare committed:
Fix unit log message (#2823)

* Fix log message for deprecated temp key

* Use string formatting and pass constant variables as arguments in log
	message to show correct name of config keys.

* Fix import order

turbokongen committed:
Exit when command_classes are missing thermostat Zwave (#2824)

nkgilley committed:
check for runtime error during db query (#2834)

shmuelzon committed:
Added support for exposing light features (#2828)

bjw-s committed:
HP ILO component (#2844)

* HP ILO component

* HP ILO component

* Add Onboard Administrator sensor

* Add Onboard Administrator sensor

* Add period to first line

Fix D400 error on line 1

DavidMStraub committed:
Add FritzBox call monitor sensor (#2791)

* Add FritzBox call monitor sensor

* Correct docstrings and suppress too few public methods warning

* Remove blank lines after docstrings

* Add blank lines after class docstrings

* Remove trailing white space

* Make daemon; add reconnect on disconnect

Commit:
Bugfix in pknx library (#2835)

* Bugfix in pknx library

* Version pinned to 0.3.3

nunofgs committed:
Add position to zwave rollershutter (#2772)

pavoni committed:
Tidy warnings.

pavoni committed:
Merge pull request #2833 from home-assistant/fix_owntracks_beacon_accuracy_bug3

Handle accuracy zero correctly in enter/leave events.

fabaff committed:
Use voluptuous for NZB sensors (#2847)

fabaff committed:
Use voluptuous for time/date sensors (#2799)

* Use voluptuous for time/date sensors

* Extend platform

* Remove additional checks

fabaff committed:
Use voluptuous for exchange sensors (#2801)

* Use voluptuous for exchange sensors

* Remove additional checks

robbiet480 committed:
Html5 notifications improvements (#2840)

* Retry sending the push for 1 day instead of failing instantly if the target is unavailable

* Add timestamp to push payload

* Correctly use the title and body fields for their intended purposes

* Add callback support

* Revert changes to frontend files.

* Add default URL which will open Home Assistant. Also put all the data into the data object of the payload so it is accessible in the browser. Without doing this, things like URL wouldnt be accessible.

* Flake8 and pylint fixes

* event->type

* Dont send the default url if actions exist

* flake8/pylint fixes again

* Update html5 tests

* Remove callbacks from this branch, will re-stage on a different branch

* Remove remnant of callbacks

* Add url to data dictionary if it exists instead of copying the entire data dictionary in

* flake8 fix

trymbill committed:
Fixed typo (#2856)

Pretty sure this is a typo ... it made finding the plist and logs a bit of a headache for me at least :)

nkgilley committed:
fix flux_update service (#2792)

* fix flux_update service

* fix tests

* give update service unique name

* remove unnecessary param

* Revert "fix tests"

This reverts commit 2fd7760455f5389b0de2e3ce272090c551d87201.

* fix flux_update

gottsman committed:
Fix logging message for disarm (#2857)

pvizeli committed:
add tcp/udp port to config validation (#2854)

robbiet480 committed:
Add Gravatar support to device_tracker (#2836)

* Support passing an email address linked to Gravatar as the picture in known_devices.

* Add a dedicated field for Gravatar

* Bring tests back up to where they were before Gravatar.

* Add tests for Gravatar.

robbiet480 committed:
Expose notify platform targets as individual services (#2837)

* First pass on providing individual services for all possible targets that a notification platform supports.

* Add a quite hacky first version of notification groups

* Add a docstring for get_targets

* Register group service under notifygroup/ and safely check for notifygroups in config

* Remove notifygroups, because it belongs in its own PR

* Make @balloob requested changes

* get_targets()->targets

* Add tests for notify targets exposed as individual services

* If we dont have a platform name set in configuration, lets use the name of the platform instead of notify

* Fix test docstring.

* Dont use a dictionary for just one value

* No need to double slugify

* targets is now just a list of strings instead of a dict

robbiet480 committed:
Add a group notify platform (#2842)

* Add a group notify platform which allows sending a single notification to multiple platforms.

* Correctly sort group.py

* Clean up the payload logic

* Make name and entity id required in the schema

* Deep update the dictionary to fix a bug where data wasnt merging.

* Add notify.group tests.

* Improve docstrings.

* Change entities to services and entity_id to service

* Make service a slug without a default value

* Update tests for entities->services, entity_id->service

* vol.Any(cv.slug) -> cv.slug

pvizeli committed:
new yahooweather version and fix update function (#2848)

balloob committed:
Add badge

balloob committed:
Add default badge to push notification (#2859)

Meatz committed:
Adding Digest Auth for webcam image retrieval (#2821)

* Adding Digest Auth for webcam image retrieval

* Update generic.py

* Update mjpeg.py

* Update generic.py

* Update mjpeg.py

* Update generic.py

* Update mjpeg.py

arsaboo committed:
Create initial Wunderground weather sensor

arsaboo committed:
Update wunderground.py

arsaboo committed:
Removed lynting issues

arsaboo committed:
Update wunderground.py

arsaboo committed:
Update wunderground.py

arsaboo committed:
Update wunderground.py

arsaboo committed:
Update wunderground.py

arsaboo committed:
Update wunderground.py

kellerza committed:
Fix script help (#2860)

Allow `--help` to filter down to the script

balloob committed:
Consider core running while starting (#2858)

DavidMStraub committed:
Added daily temp/precip forecast values to forecast.io (#2846)

Teagan42 committed:
Add imports

Teagan42 committed:
Config schema

Teagan42 committed:
Not using payload

Teagan42 committed:
Rename constant - make valid for lat/long too

Teagan42 committed:
Build url helper method

Teagan42 committed:
Unused variable

Teagan42 committed:
Pass hass to constructor

Teagan42 committed:
Validate configuration

Teagan42 committed:
Update invocation of WUndergroundData

Teagan42 committed:
Catch exception from update on initial platform setup

Teagan42 committed:
Refactoring

Teagan42 committed:
Condition is already a string

Teagan42 committed:
Don't call update on init of sensor

Teagan42 committed:
Error handling on state

Teagan42 committed:
No need no need to set variable

Teagan42 committed:
Don't set state on update - state already handles this

Teagan42 committed:
Reraise exception

Teagan42 committed:
Use url builder helper

Teagan42 committed:
Check for error and pull obvservation

Teagan42 committed:
Unit tests around wunderground

robbiet480 committed:
HTML5 notify actions (#2855)

* Add action and callback support to html5 (#2855).

Remove registrations from the callback view since we always get the latest anyway.

We dont put an audience in the claims so we will never hit this error.

Bring tests back up to where they were before callbacks.

Only import jwt where necessary

Fix bracket spacing errors

Fix JWT decode check for loop

Remove stale comment.

Add tests for the callback system.

Shorten line

Disable pylint broad-except and change e to jwt_decode_error.

Verify expiration

Remove duplicate jwt.exceptions.DecodeError

Catch no keys matched and return False

* Switch to using registrations for callbackview instead of json_path

* Only check for URL and such if the data object actually exists

* raise instead of return

* cleanup decode_jwt

* Clean up JWT errors

* Correctly set status_code to 401

* Improve JWT by adding target to claims and attempting to check the given target for a decode match first, as well as pass the target through in the event payload.

* Add tag support and fix formatting issues

* Pass through any keys that dont apply to the payload into the notification.data dictionary

* Remove stale print

* Pass back the data dictionary if it exists

* Actually put the default url even if a notify payload dictionary doesnt exist

* pylint, flake8

* Add subscription validation

* Add validation for the callback event payload and use constants where possible

* Use HTTP_UNAUTHORIZED instead of 401

* Change callback dictionary to dict instead of cv.match_all

* Fix up tests and make subscription required

* Whoops, that test was supposed to fail

* Use the result of CALLBACK_EVENT_PAYLOAD_SCHEMA as event_payload

* Add a test for html5 callback decode_jwt where the device has been renamed since notification has been sent.

* Remove the loop through logic, assume that target is always in JWT

* Always return something instead of possibly None.

* Update frontend

roidayan committed:
Add webos customize option to add custom sources (#2561)

Currently there are only hw inputs in the sources list.
Other interesting inputs can be live tv (dvbt) and vod apps.

* add customize option for webos
* add short names for livetv, youtube, mako apps
* add current app as a source
* use large icon (largeIcon is the same as icon if doesn't exists)
* filter out hw inputs that are not connected

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

Commit:
Ble tracker (#2810)

* Added Bluetooth Low Energy device tracker

* Added new file(s)

* Fixed pylint errors

* Remove traling zeros from device names

* recreated deleted file

* Added requirements

* Renamed to bluetooth_le tracker
Removed gattlib from tests
Minor code cleanup

* - fixed .coveragerc bug
- changed discovery algorithm, new devices will only be added if seen 5 times to make sure
  HA doesn't blow the database with devices just passing by

emilhetty committed:
Added name support for Forecast.io (#2638)

* Added support for name

Added name support and changed default name to "Forecast.io" since "Weather" had conflict with Yahoo weather and Open weather map

* Update forecast.py

nkgilley committed:
fix gpmdp (#2864)

* fix gpmdp

* fix balloobs comments

* move create_connection

arsaboo committed:
Updated to address @balloob's comments

arsaboo committed:
Removed blank line (linting error)

Commit:
Merge pull request #2861 from arsaboo/patch-2

Wunderground weather sensor

Teagan42 committed:
Pydoc for unit test methods

Teagan42 committed:
Remove print lines

Teagan42 committed:
Unneeded validation removed

Teagan42 committed:
Add doc link to header of file

Teagan42 committed:
Use schema validators already avaialble

Teagan42 committed:
Unused property

Teagan42 committed:
Use string constant

Teagan42 committed:
Fix config validation import to make things more readable

Teagan42 committed:
Consistent use of WUnderground

Teagan42 committed:
Use string templating

Teagan42 committed:
Extend platform schema

pvizeli committed:
add move_postion support for HA rollershutter / CONFIG_SCHEMA (#2873)

Commit:
Type-o

Commit:
Type-o

Commit:
Merge pull request #2870 from Teagan42/AddFixToWunderground

Fix PyDoc and other issues with Wunderground - approved by robbie

nkgilley committed:
fix 2862 (#2868)

balloob committed:
Fix media player art (#2879)

nkgilley committed:
binary occupancy sensor (#2869)

fabaff committed:
Use voluptuous for transport sensors (#2867)

turbokongen committed:
Migrate Thermostat and HVAC component to climate component (#2825)

* First draft for climate

* Updates for thermostats

fabaff committed:
Use voluptuous for dweet, transmission, and twitch sensor (#2802)

* Use voluptuous for dweet, transmission, and twitch sensor

* Extent platform, ordering, and consts

* Clean-up

fabaff committed:
Add url to validation (#2874)

* Add url to validation

* Fix pylint issue

* Clean-up

Commit:
Upgrade pmsensor to 0.3 (#2883)

fabaff committed:
Use voluptuous for system monitoring sensors (#2813)

* Use voluptuous for system monitoring sensors

* Extent platform, ordering, and consts

* Add resource/resources

pavoni committed:
Add voluptuous
.# Please enter the commit message for your changes. Lines starting

pavoni committed:
Change CONFIG to CONF for consistency.

pavoni committed:
Add voluptuous to Vera.

pvizeli committed:
Improve homematic climate support (#2894)

turbokongen committed:
More generic use of up and down commands plus a workaround for SOMFY controller. (#2852)

* Support older devices

* Update with workaround

* Inverted up and down

* Make dual arguments

pavoni committed:
Add voluptuous to wemo. (#2895)

technicalpickles committed:
Ensure Slack messages appear as correct user (#2893)

Current documentation suggests to use personal API tokens. This isn't
ideal because for a few reasons:

* messages will come as your own user, so it's hard to tell it's coming
  from hass
* it's harder to manage if multiple people are using Slack and home
* assistant, since you'd have to coordinate rolling of it

It is possible to use Slack bot users already. Just make a new one from https://your-team.slack.com/apps/build/custom-integration, and use the token for that. You can even add an icon from the web frontend for home assistant.

However, the message will appear as a bot without a name or icon. This pull requests fixes this by passing the as_user parameter, which uses the bot user's name and icon.

One caveat is you need to invite the bot user into the room you want to
post to. This probably was an issue before though.

:tophat: to @jnewland who pointed me to this in his branch

pavoni committed:
Merge pull request #2890 from home-assistant/loop_energy_voluptuous

Add voluptuous to Loopenergy

pavoni committed:
Tidy.

pavoni committed:
Merge pull request #2892 from home-assistant/vera_voluptuous

Add voluptuous to Vera.

persandstrom committed:
bump vsure version

persandstrom committed:
Merge pull request #2902 from persandstrom/vsure_10.1

vsure 10.1

hensing committed:
Update pymysensors version to 0.7

Commit:
Yaml secret fallback to parent folders (#2878)

* Move secret cache out of loader so it can be referenced by other folders
* Unit test to verify secrets from another folder work & see if it overrides parent secret
* Clear secret cache after load

MartinHjelmare committed:
Merge pull request #2910 from hensing/update_mysensors

Update pymysensors version to 0.7

danielperna84 committed:
Added LlamaLab Automate notify platform (#2863)

* Addod LlamaLab Automate notify platform

* Added platform to .coveragerc

* Added device-option and switched to voluptuous

* Fixed voluptuous usage

balloob committed:
Bust cache for new media player covers (#2882)

fabaff committed:
Migrate to voluptuous (#2907)

fabaff committed:
Migrate to voluptuous (#2905)

fabaff committed:
Clean-up, ordering, constants, and extend of schema (#2903)

* Clean-up, ordering, constants, and extend of schema

* Put REQUIREMENTS back and re-add line breaks

* Clean-up, ordering, constants, and extend of schema

* Extend platform

fabaff committed:
Migrate to voluptuous (#2901)

roidayan committed:
Update dht sensor dependency Adafruit_DHT to v1.3.0 (#2900)

The repository already merged the pull request adding python3 support.
root is no longer required to use the gpio.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

fabaff committed:
Use voluptuous for REST platforms (#2887)

* Initial step to migrate to voluptuous

* Migrate to voluptuous

* Add schema for sensor_classes

Commit:
Serial CO2 sensor support (#2885)

* Added support for serial HM-Z19 CO2 sensor

* Minor pylint bug fixes

* Added new files to .coveragerc

* Removed newline

* Changes in requirements after change of pmsensor library

* Change the implementation of default name

* Check if serial interface is working before adding the sensor

* Maximum sensor value is 5000ppm

mKeRix committed:
MQTT room presence detection (#2913)

* Added room presence tracker

* Fixed room/device discovery bugs

* Added tests for room tracker

* Fixed some formatting mistakes

* Fixed a tiny bug with the track new option

* Converted device tracker into sensor

* Removed leftover service entry

* Changed name to mqtt_room

* Changed payload validation to voluptuous

* Fixed validation

* Removed sleep from tests

Teagan42 committed:
Apparently, doesn't load the root config secret

Commit:
Merge pull request #2916 from Teagan42/YamlSecret-LoadBeforeBreak

Approved by @balloob via gitter

balloob committed:
Add template support to generic camera + local file tests (#2881)

* Add template support to generic camera

* Add tests for local file

balloob committed:
Update frontend

hensing committed:
Update pymysensors version to 0.7.1

MartinHjelmare committed:
Merge pull request #2917 from hensing/update_mysensors

Update pymysensors version to 0.7.1

nkgilley committed:
Add voluptuous to ecobee, speedtest.net, fast.com, actiontec, forecast.io (#2872)

* add voluptuous

* fixes for comments

* str to cv.string

balloob committed:
Update test packages (#2918)

technicalpickles committed:
Add support for notifying with Slack attachments. (#2914)

* Add support for notifying with Slack messages.

When creating notifications, this allows you to pass in `attachments`
with the `data`. It's an array of attachments as defined in
https://api.slack.com/docs/message-attachments

When passing in attachments, message is still required, but it's okay to
be a blank string.

* Split over multiple lines

* Make sure attachments gets assigned, even if there isn't attachment data

dpford committed:
Add tplink Archer C7 device tracking support for 5Ghz networks

Commit:
Add zwave.rename_node service (#2923)

* Add zwave.rename_node service

* Validate service data

* Better schema

Commit:
Merge pull request #2922 from dpford/tplink-5g

Add tplink Archer C7 device tracking support for 5Ghz networks

balloob committed:
Allow unregistering a push subscription (#2921)

* Allow unregistering a push subscription

* Update frontend

* ps - HTML5 tests DRY 🍾

balloob committed:
Update frontend

fabaff committed:
Use voluptuous for OhmConnect (#2906)

* Migrate to voluptuous

* Remove string

pavoni committed:
Add voluptuous validation to template sensor. (#2886)

fabaff committed:
Use voluptuous for dweet and arduino (#2926)

* Migrate to voluptuous

* Migrate to voluptuous

* One import is enough

fabaff committed:
Fix schemas and update ordering (#2932)

fabaff committed:
Clean-up (#2933)

pavoni committed:
Add voluptuous to template switch (#2940)

* Add voluptuous to template switch / revise tests.

pavoni committed:
Bump pywemo. (#2944)

pavoni committed:
Add voluptuous to efergy. (#2943)

kellerza committed:
Quick lint script for changed files (#2941)

Commit:
Updated python-wink version to fix color/temp detection (#2935)

kellerza committed:
Check config script (#2657)

* Add check_config, yaml linting script

* WIP: Start reusing some bootstrap methods for validation

* Start outputs

* Secrets, files and failed config

* requirements_all

* Fixes

* formatting

* Fix unit test after formatting

balloob committed:
Remove debug print

robbiet480 committed:
change const.py to use single quotes

pavoni committed:
Tidy voluptuous. (#2946)

pavoni committed:
Minor tidy of voluptuous. (#2945)

balloob committed:
Update frontend

balloob committed:
Fix broken template sensor tests

fabaff committed:
Migrate to voluptuous (#2927)

turbokongen committed:
Combine garage_door and rollershutter to cover (#2891)

* First draft for cover component

* Efficiency from @martinhjelmare

* migrate demo

* migrate demo test

* migrate command_line rollershutter

* migrate command_line test

* migrate rpi_gpio garage_door

* make some abstract methods optional

* migrate homematic

* migrate scsgate

* migrate rfxtrx and test

* migrate zwave

* migrate wink

* migrate mqtt rollershutter and test

* requirements

* coverage

* Update mqtt with garage door

* Naming and cleanup

* update test_demo.py

* update demo and core

* Add deprecated warning to rollershutter and garage_door

* Naming again

* Update

* String constants

* Make sure set_position works properly in demo too

* Make sure position is not set if not available.

* Naming, and is_closed

* Update zwave.py

* requirements

* Update test_rfxtrx.py

* fix mqtt

* requirements

* fix wink version

* Fixed demo test

* naming

fabaff committed:
Use voluptuous for Splunk (#2931)

* Migrate to voluptuous

* Update tests

fabaff committed:
Use voluptuous for graphite (#2929)

* Migrate to voluptuous

* Update tests

* Fix tests and check if Graphite instance is reachable

pvizeli committed:
add ffmpeg noise detection sensor (#2950)

nkgilley committed:
improve gpmdp (#2951)

Commit:
update frontend

Commit:
Merge remote-tracking branch 'origin/master' into dev

Commit:
pep257 fixes

robbiet480 committed:
Append the travel mode to the sensor name for Google Travel Time

freerobby committed:
Implement support for NEST structures. (#2736)

* Implement support for NEST structures.

* Conform to balloobbot style preferences.

* Log to debug level rather than info level.

* Use config validation to coerce list format if supplied as string.

* Use list comprehension for more succinct code.

* Conform to project linting standards.

BluGeni committed:
adding pull mode and relay time for you garage door (#2896)

* adding pull mode and relay time

* fixing failing tests

* removed unused vars, removed trailing whitespace

* removed white space

* split line in 2

* removed whitespace and fixed indent

* undid line break

* Update rpi_gpio.py

new line so its not too long

* back to no new line

* Moved long method to a new line

* Moved comment

* moved comment to above method

* adding required blank line

* fixed variables and made them optional

misunderstood the logic at first.

* removed line for lint and removed vars that were not required

* added second blank line for class

* added new configs to platform_schema - still have same error on load

* changing string to int

* added code to covers rpi as well

nkgilley committed:
Add volume and seek control to gpmdp (#2953)

pavoni committed:
Add voluptuous to binary template sensor (#2938)

* Add voluptuous to binary template sensor / update failing test.

* Update tests.

* Quick fixes to remove duplicate variables

Commit:
minor bug in rfxtrx sensor

turbokongen committed:
Cover (#2957)

* current_position was set, it should be optional

* Update mqtt test to match

Danielhiversen committed:
Merge pull request #2959 from home-assistant/rfxtrx

minor bug in rfxtrx sensor

turbokongen committed:
Add deprecated warning to thermostat and hvac (#2962)

* Add deprecated warning for thermostat and hvac

Commit:
Corrected sensor name from HM-Z19 to MH-Z19 (#2963)

Approved

turbokongen committed:
Bugfixing DemoCover NotImplemented Was not raised for is_closed (#2965)

* NotImplemented was not raised when is_closed was missing

* Bugfixing Demo

fabaff committed:
Migrate to voluptuous (#2958)

fabaff committed:
Migrate to voluptuous (#2955)

🐬

kellerza committed:
Check config script: various fixes (#2967)

🐬

persandstrom committed:
modbus write register service

Commit:
Add a Fan component and support for an Insteon Hub Fan (#2964)

* Fan component and service definitions
* Insteon Hub fan support

Landrash committed:
Fix reference to wrong components in tests for cameras (#1) (#2975)

🐬 👍

MartinHjelmare committed:
Extract mqtt string into constant and add log

Danielhiversen committed:
Upgrade rfxtrx lib (#2974)

pvizeli committed:
add homematic hub device with variable support / update pyhomematic with new device / add cover support (#2981)

roidayan committed:
Fix reading dht config values (#2956)

🐬

MartinHjelmare committed:
Merge pull request #2960 from MartinHjelmare/mysensors-mqtt

Add MQTT gateway for MySensors

pvizeli committed:
Full homematic system variable support (#2986)

pvizeli committed:
Change variable to poll for ccu/homegear (#2987)

nkgilley commented:
Nzbget sensor also has breaking changes.

pvizeli committed:
add motion sensor / rewrite ffmpeg binary sensor (#2969)

🐬 

* use const flags

persandstrom committed:
bump vsure version

persandstrom committed:
Merge pull request #2995 from persandstrom/vsure_10.2

bump vsure version

robbiet480 commented:
@nkgilley Can you give me a comment describing the breaking change that I can put into the notes?

nkgilley commented:
@robbiet480 

The configuration for the Sabnzbd sensor has also changed. `base_url` has been removed in favor of a required `host` and an optional `port`.  The `monitored_variables` have also changed their names to:

`article_cache`: Number of cached articles.
`average_download_rate`: Average download rate
`download_paused`: Paused downloads
`download_rate`: Current download rate
`download_size`: The size to download
`free_disk_space`: Free disk space at the storage location of NZBGet
`post_paused`: Paused posts
`remaining_size`: Remaining size to download
`uptime`: Uptime of NZBGet

fabaff committed:
Migrate to voluptuous (#2990)

🐬

robbiet480 commented:
@nkgilley These are all breaking changes or are they just deprecated? That's quite a bit of change that doesn't really belong in the release notes. Can you add a section to the NZBGet docs specifying all the changes that I can link to?

kellerza committed:
Check config requirement fix (#2999)

* Check config requirement fix

kellerza committed:
Voluptuous for pushover (#3000)

icovada committed:
Update telegram.py add send_document (#2937)

🐬

mgbowen committed:
Completely local control of entities via Alexa (#2942)

* Initial code for alexa_local_control.

* Added support for creating a dummy username.

* Move SSDP responses to local variables.

* Added config validation via Voluptuous.

* Modify and remove unnecessary returned emulated bridge values.

* Remove script and scene domains from default exposed domains.

* Replaced Flask with HomeAssistantWSGI.

* Fix lint errors.

* Correcting grammar and spelling in docs and comments.

* Rename alexa_local_control to emulated_hue.

* Rename emulated_hue attributes.

* Fix a bug where something marked not exposed is exposed by default.

* Make sure the UPNP responder thread cleanly stops when HASS stops.

Also fix some config loading and lint errors.

* Fixed unexposed entities still having individual state exposed.

* Started writing tests for emulated_hue.

* Fix being able to set state of non-exposed entity.

* Another test for emulated_hue.

* More tests for emulated_hue.

Also slightly simplified emulated_hue's PUT handler.

* Fix bad test, sorry :/

* Third time's the charm.

* Fix lint and value validation tests.

* Rename emulated_hue bridge name.

* Remove license and documentation from header.

* Combine two if statements.

* Style changes.

* Fixed various issues and added some constants

persandstrom committed:
modbus sensor value scaling (#2972)

robbiet480 committed:
Merge pull request #2970 from persandstrom/modbus_write

modbus write register service

turbokongen committed:
Use COMMAND_CLASS_THERMOSTAT_SETPOINT to get unit_of_measurement instad of COMMAND_CLASS_SENSOR_MULTILEVEL. Not all devices have multilevel sensor. (#3003)

kellerza committed:
Voluptuous for AsusWRT (#2998)

* Voluptuous for AsusWRT

MartinHjelmare committed:
Use voluptuous for mysensors (#2992)

* Add voluptuous config validation for mysensors
* Remove and clean up parts that are not needed for pymysensors 0.7.

fabaff committed:
Migrate to voluptuous (#2989)

🍪 🐬

fabaff committed:
Migrate to voluptuous (#2991)

🐬

fabaff committed:
Use voluptuous for Raspberry Pi and local file camera (#2988)

* Migrate to voluptuous

* Update const.py

* Migrate to voluptuous

* Remove duplicate _LOGGER entry

Commit:
Fan demo (#2976)

* Update attr to property and default state method

* State prop is defined in parent class

* Demo platform fan

* PyDoc

* Copy-pasta artifact

* PyDoc

* Linting

* Raise error if turn_off and turn_on not implemented

* Update demo platform

* Initial unit test commit

* Readability

* Unneeded typing

* Should inherit from fan entity

* Turn off polling

* Initial oscillating flag

* Pass HASS into demo

* Typing

* Invoke set_speed instead of setting directly

* Service update

* Update demo tests

* Forgot to block after service call.

* linting

* Test to make sure not implemented is thrown

* Is On Method test

* Update const to match string

* Update services yaml

* Toggle method

* Toggle service

* Typing

* TYPE O

* Attribute check

* Type-o

* Type-o

* Put typing back

* ToggleEntity

* Linting

* Linting

* Oops

* Stale prints

* Demo support

robbiet480 committed:
update frontend

robbiet480 committed:
Version bump to 0.27.0

robbiet480 commented:
@nkgilley Just added a note about NZBGet to the blog post telling users to go read the docs page for all the changes.

fabaff commented:
Sorry, my bad. I removed the breaking change of NZBGet along the way.
